processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java

```diff
@@ -167,12 +167,17 @@ public boolean isLimited()
       }
     }
 
+    if (!sortingNeeded) {
+      // If granularity is ALL, sortByDimsFirst doesn't change the sorting order.
+      sortingNeeded = !granularity.equals(Granularities.ALL) && sortByDimsFirst;
```

Comment:
It looks like this will change some sortByDimsFirst query executions from streaming queries into queries that need to materialize results. Is this necessary in order to fix the bug?

Reply:
Yes, `sortByDimsFirst` was not considered before. This `if` clause is to prune more cases when sorting (and materializing) is not necessary.